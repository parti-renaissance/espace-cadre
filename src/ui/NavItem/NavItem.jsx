import PropTypes from 'prop-types'
import { NavLink } from 'react-router-dom'

const NavItem = ({ path, label, icon = null, handleClick = null }) => (
  <NavLink to={path} onClick={handleClick || (() => {})} className="nav-link">
    {label}
  </NavLink>
)

NavItem.propTypes = {
  path: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  icon: PropTypes.elementType,
  handleClick: PropTypes.func,
}

export default NavItem
