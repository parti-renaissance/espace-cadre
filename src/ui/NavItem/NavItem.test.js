import { render } from '@testing-library/react'
import NavItem from 'ui/NavItem/NavItem'

jest.mock('@mui/system', () => ({
  styled: c => () => c,
}))
jest.mock('@mui/material', () => ({
  Typography: ({ children }) => <div className="mock-typography">{children}</div>,
}))

jest.mock('react-router-dom', () => ({
  NavLink: ({ children }) => <div className="mock-navlink">{children}</div>,
}))

describe('NavItem', () => {
  it('displays a NavItem', () => {
    const { container } = render(<NavItem path="path" label="foo" />)

    expect(container).toMatchSnapshot()
  })
})
