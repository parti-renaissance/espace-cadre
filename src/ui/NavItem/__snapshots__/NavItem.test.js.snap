// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem displays a NavItem 1`] = `
<div>
  <div
    class="mock-navlink"
  >
    <div
      class="mock-typography"
    >
      foo
    </div>
  </div>
</div>
`;
